src/liballoc/vec.rs

```diff
@@ -741,7 +741,7 @@ impl<T> Vec<T> {
                 return;
             }
             let remaining_len = self.len - len;
-            let s = slice::from_raw_parts_mut(self.as_mut_ptr().add(len), remaining_len);
```

Comment:
There is one more call to `drop_in_place`, for `IntoIter`. Any specific reason why you patched 2 out of 3?

Reply:
Oh, IntoIter, initial notion was that into iter would demand valid elements anyway, but if no elements are extracted from the iterator, that's not neccessarily so, so I should update the PR to include IntoIter::drop.